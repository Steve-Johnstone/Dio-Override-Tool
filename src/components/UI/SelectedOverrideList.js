import PropTypes from 'prop-types';

const SelectedOverrideList = ({ selectedOverrides, onClick }) => {
	return (
		<div>
			<p className='footer-label'>
				{selectedOverrides.length > 0 ? 'Selected Overrides: ' : ''}
			</p>
			<div className='footer-override-display'>
				<ul className='footer-override-list'>
					{selectedOverrides.map((override) => {
						return (
							<li
								key={override}
								onClick={() => onClick(override)}
								data-testid='list-item'
							>
								{override}
							</li>
						);
					})}
				</ul>
			</div>
		</div>
	);
};

SelectedOverrideList.propTypes = {
	selectedOverrides: PropTypes.array,
	onClick: PropTypes.func,
};

export default SelectedOverrideList;
