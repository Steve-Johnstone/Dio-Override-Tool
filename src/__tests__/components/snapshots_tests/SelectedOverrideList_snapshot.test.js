import renderer from 'react-test-renderer';
import SelectedOverrideList from '../../../components/UI/SelectedOverrideList';

describe('SelectedOverrideList component', () => {
	const onClick = jest.fn();

	it('should display the DOM correctly', () => {
		const tree = renderer
			.create(
				<SelectedOverrideList
					selectedOverrides={['an-override', 'second-override']}
					onClick={onClick}
				/>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
