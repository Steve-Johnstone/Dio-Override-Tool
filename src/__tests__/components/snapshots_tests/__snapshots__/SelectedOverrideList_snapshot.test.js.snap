// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedOverrideList component should display the DOM correctly 1`] = `
<div>
  <p
    className="footer-label"
  >
    Selected Overrides: 
  </p>
  <div
    className="footer-override-display"
  >
    <ul
      className="footer-override-list"
    >
      <li
        data-testid="list-item"
        onClick={[Function]}
      >
        an-override
      </li>
      <li
        data-testid="list-item"
        onClick={[Function]}
      >
        second-override
      </li>
    </ul>
  </div>
</div>
`;
